test/python/quantum_info/states/test_densitymatrix.py

```diff
@@ -359,6 +359,22 @@ def test_to_dict(self):
                     target[key] = 2 * j + i + 1
             self.assertDictAlmostEqual(target, vec.to_dict())
 
+    def test_densitymatrix_to_statevector_pure(self):
+        """Test converting a pure density matrix to statevector."""
+        state = 1/np.sqrt(2) * (np.array([1, 0, 0, 0, 0, 0, 0, 1]))
+        psi = Statevector(state)
+        rho = DensityMatrix(psi)
+        phi = rho.to_statevector()
+        self.assertAlmostEqual(psi, phi)
```

Comment:
while this  test will pass it  is not correct  due to  global phase i would would use the state fidelity